### DISCUSSION TITLE
non-standard translations

### DISCUSSION THREAD

**Discussion by ejrsimr** [GMOD/jbrowse-components#4041] [Q&A]
What is the best way to display non-standard amino acid translations when jbrowse can't automatically translate it.  e.g. selenocysteines or RNA A-to-I editing?

**Comment by cmdcolin** [comment]
@ejrsimr are you referring to the "sequence panel" e.g. here?

![image](https://github.com/GMOD/jbrowse-components/assets/6511937/0a5d67da-4399-4510-a4c0-360d60a6535b)

even if not, same idea applies to other contexts too. but, there are some interesting challenges here I'm not sure of the best way to solve all use cases yet.

currently, we basically just do very plain 3-letters of DNA == 1 protein letter, using the "standard genetic code". there are definitely exceptions like you list above. but how to integrate that into the display there is still undecided.

sort of related concepts include

- supporting non-standard genetic codes
- incorporating a genomic variant into the translation
- post translational modifications
- allowing the user to supply their own protein fasta to display instead of the one jbrowse calculates
- and indeed, like you mention, optionally incorporating other concepts like selenocysteins or external information like RNA A-to-I editing

allowing a user to supply their own protein FASTA sort of potentially "off loads the problem from jbrowse" as then someone else can do the translations as part of a data loading step. another sort of tangentially related issue is connecting the coordinates of the protein back to the genome better. this is ongoing too, and is part of our roadmap. related issues 

genome coordinates in sequence panel https://github.com/GMOD/jbrowse-components/issues/3873 
roadmap https://github.com/GMOD/jbrowse-components/issues/3974

anyways, that's just a brainstorm there. let me know if that helps or makes sense :)

**Reply by ejrsimr** [reply]
Yes, I was thinking of the sequence pane.  A user supplied FASTA is probably the most flexible.  It appears Jbrowse2 won't complain if there are stops in the CDS, which means it will show up correctly even with an odd translation and so an alternate FASTA should solve it.
